src/AppMenuSearch.cc

```
@@ -0,0 +1,427 @@
+    if (!rootMenu) {
+        return;
+    }
+
+    m_searchCandidatesDirty = false;
+    QSet<QMenu *> visited;
+    QList<QPointer<QAction>> namedAncestors;
+    QList<QPointer<QAction>> enablementAncestors;
+    collectSearchCandidates(rootMenu, visited, namedAncestors, enablementAncestors);
+}
+
+void AppMenuSearch::collectSearchCandidates(QMenu *menu, QSet<QMenu *> &visited, QList<QPointer<QAction>> &namedAncestors, QList<QPointer<QAction>> &enablementAncestors)
```

Clang (14.0.6)

[warning] method 'collectSearchCandidates' can be made static

(readability-convert-member-functions-to-static)